### DISCUSSION TITLE
Release v1.0.9

### DISCUSSION THREAD

**Discussion by BazzaCuda** [BazzaCuda/CustomMenu#24] [Announcements]
Windows 64-bit binaries

- fixed a very minor issue where consecutive right-clicks (or consecutive activations of the menu using the hotkey) without then selecting a menu item to run, resulted in a double-wide menu.

All feedback is welcome in [Discussions](https://github.com/BazzaCuda/CustomMenu/discussions).

p.s. I strongly recommend that you only download CustomMenu from my repository here on Github.

<hr /><em>This discussion was created from the release <a href='https://github.com/BazzaCuda/CustomMenu/releases/tag/v1.0.9'>Release v1.0.9</a>.</em>

**Comment by Laurens1964** [comment]
Hi BazzaCuda,

What a great piece of software you developed. I looked at many, many, many Menu apps but your is the best! Its user friendly and at the same time very flexible in adding all kind of menu items.

One small question: In which common category can i find items as "Shut down", "Restart", "Lock", etc?

Hotkey is working perfect and I experience almost no delay in bringing up submenus with a lot of items.

Thanks for this marvelous piece of software and i added a view of my main menu. Would love to see how other users are using you CusomMenu!

Laurens 
![Shot](https://github.com/user-attachments/assets/41f7d0b9-3750-4792-a7a7-dd6c28c1dedf)

**Reply by BazzaCuda** [reply]
Hi Laurens, thank you for your kind words. I'm very pleased that you find my software so useful.

To answer your questions:
Lock can be actioned by choosing "RunDLL32 Commands" from the Command Categories and "Lock PC" from the commands.
For restart and shutdown you can run shutdown.exe directly with the required commands:
Restart, **Command:** C:\Windows\System32\shutdown.exe and **Parameters:**  /r /f /t 0
Shutdown, **Command:** C:\Windows\System32\shutdown.exe and **Parameters:** /f /t 0

If there hadn't been a shutdown.exe to run, you can, of course, create a batch file to do practically anything and run the batch file from CustomMenu. In fact, using the Command and Paremeters boxes, CustomMenu can run just about anything you can type at a command prompt or in the Windows Run Box.

I would also like to know how many other people are using CustomMenu! To date, there have been at least 4,000 downloads from Github, and from MajorGeeks.com who did a feature on CustomMenu, as have other free software websites. Hopefully, all these people are using it :D

By the way, you can delete the "Right-Click me to open Config" menu item. A right-click on your main menu or on any submenu opens the config window. That menu item was an aid to getting started that I now regret adding as people seem to think it's required. It's not.

regards,
Baz

**Reply by Laurens1964** [reply]
Thanks very much BazzaCuda.
Now it looks like this.

Perhaps open an topic "show your main menu" and ask user to post a screenshot of their main menu!
![Shot](https://github.com/user-attachments/assets/5f3bf606-ff80-45a0-bf42-7f1fba80a6a2)

**Reply by BazzaCuda** [reply]
What a good idea. Done!

**Comment by Laurens1964** [comment]
Hi Baz,

Would you consider an option to change the background color of CustomMenu?
I love to incorporate CustumMenu in my Win11 colorscheme. I don't dislike the black background but it is "not in sync" with my colorscheme.

Thanks for considering this.

Laurens

**Reply by BazzaCuda** [reply]
I'll have a think about that. I certainly don't have any objections. In fact, I recently suggested a straight-forward way (with a colors.ini file) of providing color customization to the author of Free File Sync. Maybe now would be a good time to implement my own suggestion :P

**Reply by Laurens1964** [reply]
Hi Baz,

Any ideas for a next release? Or to busy with other software development?
My short list would be:

1. Copy a couple of standard icon DLL's to the CustomMenu directory during install
2. Option to change background color of the menu
3. Automatic backup of the CustomMenu.ini to a CustomMenu.bak.

I'm using your CustomMenu daily and still discovering a ton of options in the command categories and common shell folders.
So again thanks from a "heavy user" :-)

**Reply by BazzaCuda** [reply]
Hi Laurens,

I'm currently engaged on other projects at the moment and even then, my time on them is very limited.
As a result, I'm only considering bug fixes just now and in strict order of severity across all my projects.

A quick fix to save you having to navigate to some of the common icon DLLs might be to have a text file containing the full paths to your DLLs. You can then copy and paste the DLL path from the text file into the Icon File box. Just a thought.

If I get chance to do some actual Custom Menu development, I will probably tackle them in 2, 3, 1 order (your numbers).

**Reply by BazzaCuda** [reply]
> Hi Baz,
> 
> Any ideas for a next release? Or to busy with other software development? My short list would be:
> 
> 1. Copy a couple of standard icon DLL's to the CustomMenu directory during install
> 2. Option to change background color of the menu
> 3. Automatic backup of the CustomMenu.ini to a CustomMenu.bak.
> 
> I'm using your CustomMenu daily and still discovering a ton of options in the command categories and common shell folders. So again thanks from a "heavy user" :-)

Hi Laurens,

I don't know if you'll get this message, but we'll try.

I have finally managed to find some time to do a bit of development with CustomMenu and I have today added your request to be able change the colors.
If you're interested in trying out a pre-release copy I'll be happy to share it with you here to get your feedback.

There are currently three colors you can customize via the new "colors.ini" file:
background color: for the menus, the config dialog and the icon explorer.
highlight color: for the menu items as you hover over them
info color: for the info text on the config dialog and icon explorer (e.g. Help and Feedback) and the submenu chevron in the menus:
![image](https://github.com/user-attachments/assets/d358efb9-3265-4730-8225-684981ba18ab)


I look forward to hearing from you.

Baz

p.s. It also makes a backup copy of your CustomMenu.ini file before each save.

**Reply by Laurens1964** [reply]
Hi Baz,

I will try to have a look at it next week. 
To be honest I'm not using CustomMenu anymore because of a lack of development. But now you have a new release I will have a look at it. So please send me a pre release that i can test.

Thanks for you support.

Laurens
